src/engraving/libmscore/note.cpp

```diff
@@ -2322,6 +2322,16 @@ bool Note::dotIsUp() const
     }
 }
 
+static bool hasAlteredUnison(Note* note)
+{
+    const auto& chordNotes = note->chord()->notes();
+    AccidentalVal accVal = tpc2alter(note->tpc());
+    int relLine = absStep(note->tpc(), note->epitch());
+    return std::find_if(chordNotes.begin(), chordNotes.end(), [note, accVal, relLine](Note* n) {
+        return n != note && !n->hidden() && absStep(n->tpc(), n->epitch()) == relLine && tpc2alter(n->tpc()) != accVal;
+    }) != chordNotes.end();
```

Comment:
i.e. look for a note on the same A-G step (including octave), but that would require a different accidental.